src/ZeroCrossImpl.java

```diff
@@ -0,0 +1,177 @@
+import java.util.Scanner;
+
+public class ZeroCrossImpl implements ZeroCross {
+
+    int whoPlays;
+    public int firstPlayer;
+    public int secondPlayer;
+
+
+    @Override
+    public void run() {
+
+
+        Scanner scanner = new Scanner(System.in);
+
+        int[][] zeroCross = new int[3][3];
+
+        System.out.println("Введите целые числа которыми будете играть в игру Крестик-Нолик" + " " + "роль нолика занимает число" +
+                "-" + "x");
+
+        int x = scanner.nextInt();
+        firstPlayer = x;
+
+        System.out.println(" А роль крестика занимает число" + "-" + "y");
+
+        int y = scanner.nextInt();
+        secondPlayer = y;
+        System.out.println("игру начинает:"+ "x");
+
+
+        while (!(IsWinerZero(zeroCross) || IsWinerCross(zeroCross))) {
+
+
+            System.out.println("Введите число от 0 до 2 для позиции по горизонтали");
+            int line = scanner.nextInt();
+
+            System.out.println("Введите число от 0 до 2 для позиции по вертикали");
+            int column = scanner.nextInt();
+
+
+
+            if (zeroCross[line][column] == 0) {
+                zeroCross[line][column] = whoNext();
+            } else {
+                System.out.println("Вы не можите играть в этой позиции, введите еще раз");
+            }
+
+            System.out.println("Игровое поле:");
+            System.out.println(" В игру вступает: "+ whoPlays);
+
+            for (int i = 0; i < zeroCross.length; i++) {
+                for (int j = 0; j < zeroCross[i].length; j++) {
+                    System.out.print(zeroCross[i][j] + "\t");
+                }    // zeroCross[i][j]=(i+1)*(j+1);
+                System.out.println();
+            }
+
```

Comment:
Output of game board might be just another method. Then you clear up and do it understandable.